src/wallet/test/coinselector_tests.cpp

```diff
@@ -453,6 +456,37 @@ BOOST_AUTO_TEST_CASE(bnb_search_test)
     }
 }
 
+BOOST_AUTO_TEST_CASE(tx_creation_bnb_sffo_restriction)
+{
+    // Verify the transaction creation process does not produce a BnB solution when SFFO is enabled.
+    // This is currently problematic because it could require a change output. And BnB is specialized on changeless solutions.
+    std::unique_ptr<CWallet> wallet = NewWallet(m_node);
+    WITH_LOCK(wallet->cs_wallet, wallet->SetLastBlockProcessed(300, uint256{})); // set a high block so internal UTXOs are selectable
+
+    CTxDestination change_dest = *Assert(wallet->GetNewChangeDestination(OutputType::BECH32));
+    CTxOut change_out(0, GetScriptForDestination(change_dest));
+    size_t change_output_size = GetSerializeSize(change_out);
+
+    CFeeRate effective_feerate(10000);
+    CFeeRate discard_feerate = GetDiscardRate(*wallet);
+
+    CAmount change_fee = effective_feerate.GetFee(change_output_size);
+    size_t change_spend_size = CalculateMaximumSignedInputSize(change_out, wallet.get(), nullptr);
+    CAmount cost_of_change = discard_feerate.GetFee(change_spend_size) + change_fee;
+
+    // Add spendable coin at the BnB selection upper bound
+    CoinsResult available_coins;
+    add_coin(available_coins, *wallet, COIN + cost_of_change, /*feerate=*/CFeeRate(0), /*nAge=*/6*24, /*fIsFromMe=*/true, /*nInput=*/0, /*spendable=*/true);
+
+    // Now verify coin selection does not produce BnB result
+    ASSERT_DEBUG_LOG_NOT_FOUND("Coin Selection: Algorithm:bnb");
```

Comment:
I actually did it in this way per achow request during coredev. My initial version involved returning the coin selection information in a separate struct https://github.com/furszy/bitcoin-core/commit/a1b2c53d1feddf748d7d91a884fe52e50998f61e in the tx creation result.

I do prefer my previous approach because it would allows us to retrieve this information at the RPC level as well. However, I am also fine going with this other option as well. It's not something widely used nowadays anyway. We can also revisit this when another feature/test needs it.

Reply:
For end users, it may not be particularly useful. However, I think that additional information at the RPC level could be beneficial for other utilities, like a coin selection metrics tool or for connected applications that uses core as their backend.
That said, there's no need to add it unless there's a need for it.

Other than that, will return to the struct commit then.